### DISCUSSION TITLE
Spark Doris Connector ：does not allow insertion

### DISCUSSION THREAD

**Discussion by 666chaoge666** [apache/doris#7200] [A - General / Q&A]
Spark Doris Connector在进行select操作时，可正常返回数据，但insert时报解析错误，查看了源码，Relation不为InsertIntoTable导致的，但不清楚问题原因和如何解决，求大神帮忙！
scala> spark.sql("""
     | INSERT INTO spark_doris3 VALUES ("test4","test_imei","test_uuid",20211101,20211123)
     | """)
org.apache.spark.sql.AnalysisException: org.apache.doris.spark.sql.DorisRelation@5b61d156 does not allow insertion.;;
'InsertIntoTable Relation[apk_name#10,imei#11,uuid#12,first_time#13L,time#14L] org.apache.doris.spark.sql.DorisRelation@5b61d156, false, false

![image](https://user-images.githubusercontent.com/77181242/142989757-81597644-1a39-4af0-ad9a-a6906defaca4.png)

**Comment by tinkerrrr** [comment]
拉master分支的代码，打包spark-doris-connector就行。如果要用在0.14版本的doris上，还得把 https://github.com/apache/incubator-doris/blob/master/extension/spark-doris-connector/src/main/java/org/apache/doris/spark/DorisStreamLoad.java#L78 这里改成用 `RestService.randomBackend`
